torch/distributions/distribution.py

```diff
@@ -258,9 +265,14 @@ def _validate_sample(self, value):
             if i != 1 and j != 1 and i != j:
                 raise ValueError('Value is not broadcastable with batch_shape+event_shape: {} vs {}.'.
                                  format(actual_shape, expected_shape))
-        assert self.support is not None
-        if not self.support.check(value).all():
-            raise ValueError('The value argument must be within the support')
+        try:
+            assert self.support is not None
+            if not self.support.check(value).all():
+                raise ValueError('The value argument must be within the support')
+        except NotImplementedError:
+            warnings.warn(f'{self.__class__} does not define `support` to enable ' +
+                          'sample validation. Please initialize the distribution with ' +
+                          '`validate_args=False` to turn off validation.')
```

Comment:
You might make this a little more targeted as
```py
try:
    support = self.support
except NotImplementedError:
    warnings.warn(...)
assert support is not None
if not support.check(value).all():
    ...
```

Reply:
Thanks, done.